doc/basic_types.md

```diff
@@ -1,5 +1,409 @@
+otherwise.
+
+#### IsDataView
+```cpp
+bool IsDataView() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `DataView` or `false`
+otherwise.
+
+#### IsBuffer
+```cpp
+bool IsBuffer() const;
+```
+
+Returns `true` if the underlying value is a Node `Buffer` or `false`
+otherwise.
+
+#### IsExternal
+```cpp
+bool IsExternal() const;
+```
+
+Returns `true` if the underlying value is a N-API external object or `false`
+otherwise.
+
+#### ToBoolean
+```cpp
+Boolean ToBoolean() const;
+```
+
+Returns a `Napi::Boolean` representing the `Napi::Value`.
+
+This is a wrapper around `napi_coerce_to_boolean`. This will throw a JavaScript
+exception if the coercion fails. If C++ exceptions are not being used, callers
+should check the result of `Env::IsExceptionPending` before attempting to use
+the returned value.
+
+
+#### ToNumber
+```cpp
+Number ToNumber() const;
+```
+
+Returns a `Napi::Number` representing the `Napi::Value`.
+
+Note:
+This can cause script code to be executed according to JavaScript semantics. 
+This is a wrapper around `napi_coerce_to_number`. This will throw a JavaScript
+exception if the coercion fails. If C++ exceptions are not being used, callers
+should check the result of `Env::IsExceptionPending` before attempting to use
+the returned value.
+
+
+#### ToString
+```cpp
+String ToString() const;
+```
+
+Returns a `Napi::String` representing the `Napi::Value`.
+
+Note that this can cause script code to be executed according to JavaScript
+semantics. This is a wrapper around `napi_coerce_to_string`. This will throw a
+JavaScript exception if the coercion fails. If C++ exceptions are not being
+used, callers should check the result of `Env::IsExceptionPending` before
+attempting to use the returned value.
+
+
+#### ToObject
+```cpp
+Object ToObject() const;
+```
+
+Returns a `Napi::Object` representing the `Napi::Value`.
+
+This is a wrapper around `napi_coerce_to_object`. This will throw a JavaScript
+exception if the coercion fails. If C++ exceptions are not being used, callers
+should check the result of `Env::IsExceptionPending` before attempting to use
+the returned value.
+
+
+## Name
+
+Names are JavaScript values that can be used as a property name. There are two
+specialized types of names supported in Node Addon API- [`String`](String.md)
+and [`Symbol`](Symbol.md).
+
+### Methods
+
+#### Constructor
+```cpp
+Name();
+```
+
+Returns an empty `Name`.
+
+```cpp
+Name(napi_env env, napi_value value);
+```
+- `[in] env` - The environment in which to create the array.
+- `[in] value` - The primitive to wrap.
+
+Returns a Name created from the JavaScript primitive.
+
+Note:
+The value is not coerced to a string.
+
+## Array
+
+Arrays are native representations of JavaScript Arrays. `Napi::Array` is sugar
+around `napi_value` representing a JavaScript Array.
+
+### Constructor
+```cpp
+Array();
+```
+
+Returns an empty array.
+
+If an error occurs, a `Napi::Error` will get thrown. If C++ exceptions are not
```

Comment:
"will *be* thrown" might be better – here, and subsequently.

Reply:
+1